cipher/transposition/transposition_test.go

```diff
@@ -101,3 +101,45 @@ func TestEncryptDecrypt(t *testing.T) {
 		t.Error("The string was decrypted with a different key")
 	}
 }
+
+func FuzzTransposition(f *testing.F) {
+	for _, transpositionTestInput := range getTexts() {
+		f.Add(transpositionTestInput)
+	}
+	f.Fuzz(func(t *testing.T, input string) {
+		keyword := getRandomString()
+		encrypted, err := Encrypt([]rune(input), keyword)
+		switch {
+		case err == nil:
+		case errors.Is(err, &KeyMissingError{}),
+			errors.Is(err, &NoTextToEncryptError{}):
+			return
+		default:
+			t.Fatalf("unexpected error when encrypting string %q: %v", input, err)
+		}
+		decrypted, err := Decrypt([]rune(encrypted), keyword)
+		switch {
+		case err == nil:
+		case errors.Is(err, &KeyMissingError{}),
+			errors.Is(err, &NoTextToEncryptError{}):
+			return
+		default:
+			t.Fatalf("unexpected error when decrypting string %q: %v", encrypted, err)
+		}
+		if !runesStartsWith([]rune(decrypted), []rune(input)) {
+			t.Fatalf("expected: %+v, got: %+v", []rune(input), []rune(decrypted))
+		}
+	})
+}
+
+func runesStartsWith(s, start []rune) bool {
+	if len(start) == 0 {
+		return true
+	}
```

Comment:
😕

A simple `string(decrypted) == input` might suffice no?

Reply:
Unfortunately there are two issues about using it:
The first one is that the decrypted string contains whitespaces after the end of the word (I'm adding an example at the end).

The second reason is that the fuzzer complains when it starts encoding special chars, i.e. [65533](https://www.compart.com/en/unicode/U+FFFD)

Example, when comparing using this code

```go
if !strings.Contains(string(decrypted), input) {
	t.Fatalf("expected: %+v, got: %+v", []rune(input), []rune(decrypted))
}
```

I get this result when launching the fuzzer

```bash
go test ./cipher/transposition -fuzz=FuzzTransposition
fuzz: elapsed: 0s, gathering baseline coverage: 0/139 completed
failure while testing seed corpus entry: FuzzTransposition/deaae2ca3cf382f428a20635cec48a44d2ff6a25a92dd36639ac7d49a542b8c6
fuzz: elapsed: 0s, gathering baseline coverage: 4/139 completed
--- FAIL: FuzzTransposition (0.02s)
    --- FAIL: FuzzTransposition (0.00s)
        transposition_test.go:130: expected: [65533], got: [65533 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32 32]
```

Reply:
> The first one is that the decrypted string contains whitespaces after the end of the word (I'm adding an example at the end).

Then isn't that a `bug` rather than something that this test should accommodate?

Reply:
Yeah, it was a weird behavior to me when I discovered it the first time. At first I thought to keep this behavior as it was implemented in this way in the first place and the existing unit tests are accommodating it, but not that we are discussing about it I feel more comfortable to change and fix this thing. Thanks for the feedback, I'll work on it